data-prepper-plugins/key-value-processor/src/test/java/com/amazon/dataprepper/plugins/processor/keyvalue/KeyValueProcessorTests.java

```diff
@@ -97,22 +100,97 @@ void testSingleRegexKvDelimiterKvToObjectKeyValueProcessor() {
 
         final Record<Event> record = getMessage("key1:++:value1&key2:+:value2");
         final List<Record<Event>> editedRecords = (List<Record<Event>>)keyValueProcessor.doExecute(Collections.singletonList(record));
-        LinkedHashMap parsed_message = getLinkedHashMap(editedRecords);
+        final LinkedHashMap parsed_message = getLinkedHashMap(editedRecords);
 
         assertThat(parsed_message.size(), equalTo(2));
         assertThatKeyEquals(parsed_message, "key1", "value1");
         assertThatKeyEquals(parsed_message, "key2", "value2");
     }
 
+    @Test
+    void testBadKeyValueDelimiterRegexKeyValueProcessor() {
+        when(mockConfig.getKeyValueDelimiterRegex()).thenReturn("[");
+        try {
```

Comment:
Please use `assertThrows` instead.

It should look like:

```
PatternSyntaxException ex = assertThrows(PatternSyntaxException.class, () -> new KeyValueProcessor(pluginMetrics, mockConfig));
assertThat(e.getMessage(), startsWith("key_value_delimiter_regex"));
```

Hamcrest provides a `startsWith` (or maybe its `beginsWith`) to verify that a string starts with a given string. This produces a more helpful output when it fails because you can see the string.

Please change all the other tests as well.